shared_model/validators/abstract_validator.hpp

```diff
@@ -14,6 +15,12 @@ namespace shared_model {
     // validator which can be overloaded for dynamic polymorphism
     template <typename Model>
     class AbstractValidator {
+     protected:
+      bool validateHexString(const std::string &str) const {
```

Comment:
This does not seem to be a fitting place for this function, maybe move it out of class?